cocotbext/axi/axis.py

```diff
@@ -95,6 +100,11 @@ def normalize(self):
         else:
             self.tkeep = [1]*n
 
+        if self.tstrb is not None:
+            self.tstrb = self.tstrb[:n] + [self.tstrb[-1]]*(n-len(self.tstrb))
+        else:
+            self.tstrb = self.tkeep
```

Comment:
This is deliberate. If `tstrb` is not used, it should match `tkeep`. This implies 2 possible states: 

  - keep byte + byte is data
  - may discard byte
  
It also means the invalid state of `tkeep` low + `tstrb` high is avoided.